### PR TITLE
Add support for i18n

### PR DESCRIPTION
**Pull request by harshithpabbati**
Screenshot 2020-05-08 at 5 35 59 PM

Thanks @Xambey for your [PR](https://github.com/graphql/graphiql/pull/977).

### CODE REVIEW

**Comment by harshithpabbati** [comment]
@acao here's the PR :)

File: packages/graphiql/tsconfig.json [hunk 1]
```diff
@@ -8,11 +8,12 @@
     "baseUrl": ".",
     "target": "es5",
     "strictPropertyInitialization": false,
+    "resolveJsonModule": true,
```
Comment (acao) [inline packages/graphiql/tsconfig.json]: lets add this to the base config

File: packages/graphiql/src/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import i18n from 'i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+import { initReactI18next } from 'react-i18next';
+
+const enTranslations = require('./locales/en/translation.json');
```
Comment (acao) [inline packages/graphiql/src/i18n.ts]: are we able to use `import enTranslations from` or `import()` here for better tree shaking?

File: packages/graphiql/src/components/ToolbarButton.tsx [hunk 3]
```diff
@@ -33,13 +34,14 @@ export class ToolbarButton extends React.Component<
...
+    const { t } = this.props;
     return (
       <button
         className={'toolbar-button' + (error ? ' error' : '')}
         onClick={this.handleClick}
-        title={error ? error.message : this.props.title}
+        title={t(error ? error.message : this.props.title)}
         aria-invalid={error ? 'true' : 'false'}>
-        {this.props.label}
+        {t(this.props.label)}
```
Comment (acao) [inline packages/graphiql/src/components/ToolbarButton.tsx]: hmm.. not sure if its more efficient to translate it when passing the prop, or at render. what do you think?

File: packages/graphiql/tsconfig.json (hunk 1 above)
Comment (harshithpabbati) [reply packages/graphiql/tsconfig.json]: okay

File: packages/graphiql/src/components/ToolbarButton.tsx (hunk 3 above)
Comment (harshithpabbati) [reply packages/graphiql/src/components/ToolbarButton.tsx]: Yeah, it translates once it receives prop

File: packages/graphiql/src/components/DocExplorer.tsx [hunk 4]
```diff
@@ -73,7 +74,7 @@ export class DocExplorer extends React.Component<
   }
 
   render() {
-    const { schema } = this.props;
+    const { schema, t } = this.props;
```
Comment (acao) [inline packages/graphiql/src/components/DocExplorer.tsx]: this component is being moved to a new package, however I *think* these changes will follow without much conflict when we merge all the moved indexes

**Comment by harshithpabbati** [comment]
@acao made the required changes

**Comment by harshithpabbati** [comment]
Created a new [PR](https://github.com/graphql/graphiql/pull/1526) in GraphiQL repo, so closing this.

**Pull request closed without merging**